### PR TITLE
User template blocks to allow better extensibility

### LINKED ISSUE #4949: Improve extensibility of the 'user form' template
### CKAN Version if known (or site URL)

2.8.x

### Please describe the expected behaviour

As a developer of extensions, I expect to extend/override specific parts of the template at `ckan/templates/user/edit_user_form.html` be possible and easy, without needing to copy the whole original code. 

For instance, in the context of a deploy that uses LDAP as access protocol, I'd like to hide/remove the password related fieldset because it has no point. 

### Please describe the actual behaviour

It is not possible to extend  because there is no proper blocks

### PR DESCRIPTION
**Pull request by Gauravp-NEC**
Fixes #4949

### Proposed fixes
This PR fixes issue https://github.com/ckan/ckan/issues/4949
1. Added few block tags.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/templates-bs3/user/edit_user_form.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {% import 'macros/form.html' as form %}
 
-{% block form %}
+{% block user_edit_form %}
```
Comment (amercader) [inline ckan/templates-bs3/user/edit_user_form.html:3]: Do not change existing block names as it might break extensions

File: ckan/templates-bs3/user/edit_user_form.html [hunk 2]
```diff
@@ -69,16 +71,16 @@
     {% endblock %}
     {% endif %}
 
+    {% block form_actions %}
     <div class="form-actions">
-      {% block form_actions %}
```
Comment (amercader) [inline ckan/templates-bs3/user/edit_user_form.html:73]: Also don't change the scope of existing blocks for the same reason, this particular one seems fine as is

File: ckan/templates-bs3/user/edit_user_form.html (hunk 1 above)
Comment (Gauravp-NEC) [reply ckan/templates-bs3/user/edit_user_form.html]: Reverted it back to `{% block form %}`

File: ckan/templates-bs3/user/edit_user_form.html (hunk 2 above)
Comment (Gauravp-NEC) [reply ckan/templates-bs3/user/edit_user_form.html]: Done

**Comment by Gauravp-NEC** [comment]
@amercader , I think the test failure is unrelated to this PR.

File: ckan/templates-bs3/user/edit_user_form.html [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 {% block form %}
   <form id="user-edit-form" class="dataset-form" method="post" action="{{ action }}" enctype="multipart/form-data">
-    {{ form.errors(error_summary) }}
+    {% block form_errors %}{{ form.errors(error_summary) }}{% endblock %}
 
-    {% block core_fields %}
+    {% block details_fieldset %}
```
Comment (amercader) [inline ckan/templates-bs3/user/edit_user_form.html:7]: As I said, don't change existing block names, please check all instances

Comment (Gauravp-NEC) [reply ckan/templates-bs3/user/edit_user_form.html]: Reverted

**Comment by Gauravp-NEC** [comment]
@amercader , Now all checks have been passed. Please let me know if anything is left to do in this PR. Thanks :)

**Comment by Gauravp-NEC** [comment]
@amercader , I think `Changelog entries / towncrier (pull_request)`  failure is not related to this PR. Please let me know your opinion. Thanks :)

**Comment by Gauravp-NEC** [comment]
@amercader , I think `Changelog entries / towncrier` failure is not related to this PR.

File: ckan/tests/lib/search/test_query.py [hunk 4]
```diff
@@ -353,7 +353,7 @@ def test_name_multiple_results(self):
         factories.Dataset(name="second-record")
         factories.Dataset(name="third-dataset")
         result = search.query_for(model.Package).run({"q": u"record"})
-        assert set(result["results"]) == {"first-record", "second-record"}
+        assert set(result["results"]) == {"second-record", "first-record"}
```
Comment (amercader) [inline ckan/tests/lib/search/test_query.py:356]: Please don't include unrelated changes. If there is an issue with this test it will be fixed in a separate PR

Comment (Gauravp-NEC) [reply ckan/tests/lib/search/test_query.py]: > Please don't include unrelated changes. If there is an issue with this test it will be fixed in a separate PR

@amercader , I have revert back the unrelated changes. Thanks for the review  : )
